### DISCUSSION TITLE
PR descriptions are cut off

### DISCUSSION THREAD

**Discussion by papermana** [renovatebot/renovate#29960] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

Sometimes descriptions of PRs opened by Renovate are cut off in the middle. I've seen this multiple times, but haven't paid much attention to it, so I haven't found a pattern to when it happens. However, it seems like it only happens in PRs that update a lot of dependencies. Usually, the description is cut off in the middle of the changelog list, but right now it happened in the middle of the table of updated dependencies. This means that the PR can't even be rebased (without going to developer.mend.io). For context, we update all minor/patch dependencies at the same time, so there's 86 (unique) dependencies in this PR (100+ including duplicates). It looks like this:

<img width="835" alt="Screenshot 2024-07-01 at 10 57 15" src="https://github.com/renovatebot/renovate/assets/10849003/b0c08238-8aea-40cb-a9c0-414d73b76f45">

UPDATE: I looked at the description of the most recent PR where this occurs (in markdown mode) and it happens to contain exactly 60,000 characters. Is there some limit?

### Logs (if relevant)

<details><summary>Logs</summary>
I think this is the relevant part; however, I don't see any issues.

```

DEBUG: Preparing files for committing to branch dependencies/all-non-major
DEBUG: Setting git author name: renovate[bot]
DEBUG: Setting git author email: 29139614+renovate[bot]@users.noreply.github.com
DEBUG: git commit
DEBUG: HEAD https://api.github.com/repos/cosuno/cosuno/git/refs/heads/dependencies/all-non-major/ = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=256)
DEBUG: HEAD https://api.github.com/repos/cosuno/cosuno/git/refs/heads/dependencies/all-non-major = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=217)
DEBUG: resetToCommit(c8adcdf8e1226f1dce576bc1f10d51ad6e93efea)
DEBUG: Fetching branch dependencies/all-non-major
DEBUG: Setting current branch to master
DEBUG: latest commit
INFO: Branch created
DEBUG: Updating status check state to failed
DEBUG: Setting branch status
DEBUG: http cache: saving https://api.github.com/repos/cosuno/cosuno/commits/dependencies/all-non-major/status (etag=W/"5f2a983282cbc7eaaaa05e7cc6bedcb337f290cb2ad00bbfd212f097d1dfbfd3", lastModified=undefined)
DEBUG: Updating renovate/stability-days status check state to green
DEBUG: Setting branch status
DEBUG: http cache: saving https://api.github.com/repos/cosuno/cosuno/commits/dependencies/all-non-major/status (etag=W/"61ddf4f8e1b327434c1cc966175520e6d51e2e5ed50cb265ea6300cee6dd9e81", lastModified=undefined)
DEBUG: Ensuring PR
DEBUG: There are 0 errors and 0 warnings
DEBUG: getBranchPr(dependencies/all-non-major)
DEBUG: findPr(dependencies/all-non-major, undefined, open)
DEBUG: findPr(dependencies/all-non-major, undefined, closed)
DEBUG: getPrCache()
DEBUG: Forcing PR because of artifact errors
# [snipped list of all dependencies but they looked like the line below]
DEBUG: Fetching changelog: https://github.com/Microsoft/TypeScript (5.4.5 -> 5.5.2)
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Removing duplicate release notes
DEBUG: Creating PR
DEBUG: Creating PR
DEBUG: PR created
DEBUG: Adding labels 'Crane, Steel, 📦 dependencies' to #10517
INFO: PR created

```

</details>

**Comment by rarkins** [comment]
Yes, GitHub imposes the character limit. Can you look at the source of the markdown for this PR and determine if there's any "inefficient" text which could be removed or shortened? This is the first time I've seen truncation occur in the table itself - normally it's the release notes which are truncated.

**Reply by papermana** [reply]
I don't see any one thing that would be particularly long. All the badges add up, though. Maybe in case a generated description has over 60k characters, you could produce a minimalist one instead, that would contain only a list of dependencies with links to their repos and version numbers.

**Reply by rarkins** [reply, marked as answer]
You can modify `prBodyColumns`. I suggest you remove Age, Passing and Adoption and just leave Confidence

**Reply by HonkingGoose** [reply]
> Maybe in case a generated description has over 60k characters, you could produce a minimalist one instead, that would contain only a list of dependencies with links to their repos and version numbers.

That would mean putting _each_ platform's character limits in the code, and writing a new code path to shrink the PR body text when the text goes over the limit.

I'll let the maintainers decide if that's a feature they want to have.

**Reply by rarkins** [reply]
We do have that concept today, but we just remove content instead of adjusting it. The effort to intelligently shrink/reduce it would be quite high, instead I recommend the user removes columns themselves for this edge case
